src/coriolis.jl

```diff
@@ -1,45 +1,104 @@
+using .TurbulenceClosures: ▶xy_cfa, ▶xy_fca
+
 #####
 ##### Functions for non-rotating models
 #####
 
-@inline x_f_cross_U(i, j, k, grid::AbstractGrid{T}, ::Nothing, U) where T = zero(T)
-@inline y_f_cross_U(i, j, k, grid::AbstractGrid{T}, ::Nothing, U) where T = zero(T)
-@inline z_f_cross_U(i, j, k, grid::AbstractGrid{T}, ::Nothing, U) where T = zero(T)
+@inline x_f_cross_U(i, j, k, grid::AbstractGrid{FT}, ::Nothing, U) where FT = zero(FT)
+@inline y_f_cross_U(i, j, k, grid::AbstractGrid{FT}, ::Nothing, U) where FT = zero(FT)
+@inline z_f_cross_U(i, j, k, grid::AbstractGrid{FT}, ::Nothing, U) where FT = zero(FT)
 
 #####
 ##### The 'FPlane' approximation. This is equivalent to a model with a constant
 ##### rotation rate around its vertical axis.
 #####
 
 """
-    FPlane{T} <: AbstractRotation
+    FPlane{FT} <: AbstractRotation
 
 A parameter object for constant rotation around a vertical axis.
 """
-struct FPlane{T} <: AbstractRotation
-    f :: T
+struct FPlane{FT} <: AbstractRotation
+    f :: FT
 end
 
 """
-    FPlane([T=Float64;] f)
+    FPlane([FT=Float64;] f=nothing, rotation_rate=nothing, latitude=nothing)
 
 Returns a parameter object for constant rotation at the angular frequency
-`2f`, and therefore with background vorticity `f`, around a vertical axis.
+`f/2`, and therefore with background vorticity `f`, around a vertical axis.
+If `f` is not specified, it is calculated from `rotation_rate` and
+`latitude` according to the relation `f = 2*rotation_rate*sind(latitude).
 
-Also called `FPlane`, after the "f-plane" approximation for the local effect of 
+Also called `FPlane`, after the "f-plane" approximation for the local effect of
 Earth's rotation in a planar coordinate system tangent to the Earth's surface.
 """
-function FPlane(T::DataType=Float64; f) 
-    return FPlane{T}(f)
+function FPlane(FT=Float64; f=nothing, rotation_rate=nothing, latitude=nothing)
+
+    if f==nothing && rotation_rate != nothing && latitude != nothing
+        return FPlane{FT}(2rotation_rate*sind(latitude))
+    elseif f != nothing && rotation_rate == nothing && latitude == nothing
+        return FPlane{FT}(f)
```

Comment:
What's the benefit of using `isnothing`, just curious.

Also, would you recommend using `!isnothing` instead of `f != nothing`?

Reply:
It probably doesn't matter in this case as we're comparing simple stuff, but apparently `f == nothing` can have a performance hit and `f === nothing` is a slightly better way of checking for this. But instead of always remembering to use `===`, there's an `isnothing` function we can use which is why I started using it.

See: https://github.com/JuliaLang/julia/issues/29674 and https://github.com/JuliaLang/julia/pull/29679

But this isn't important for this PR. I'm happy with `f == nothing` in this case.

Reply:
I guess if we use `isnothing()` then `!isnothing` is more consistent, although `f !== nothing` might be clearer. This PR is fine as it doesn't use `isnothing` with `!== nothing` together.

Maybe this is getting too pedantic.

Reply:
Okay, so I'm leaving it as it is.